NOTICE.txt

```diff
@@ -0,0 +1,8 @@
+Apache MINA SSHD
+Copyright 2008-2018 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+Please refer to each LICENSE.<component>.txt file for the
```

Comment:
Hmm? Maybe it should not be in this repo

Reply:
See https://github.com/jenkinsci/sshd-module/pull/24#discussion_r207382133, I thought it may be required since I am copying an Apache-licensed source file, see [the Apache License 2.0 section 4d](http://www.apache.org/licenses/LICENSE-2.0?#redistribution). I am certainly no expert on the subject though.

Reply:
I'm not an expert, but I do sometimes review releases at Apache Incubator where this question comes up far more often. The NOTICE file is for adding things licensed under ALv2 (and possibly other licenses) where you've copied differently licensed code into your project. One Apache project including another is pretty simple, but it gets more complicated with the notice file when you start including other code that has an attribution clause like in ALv2.

Reply:
See [here](https://www.apache.org/dev/licensing-howto.html#permissive-deps)